azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java

```diff
@@ -78,59 +81,75 @@ private IWebApp getWebApp(final WebAppConfig config) {
-        final ResourceGroup resourceGroup = null;
+        // todo: Extract resource group logic to library
+        ResourceGroup resourceGroup = az.getAzureResourceManager().resourceGroups().getByName(webAppConfig.getResourceGroup());
+        if (resourceGroup == null) {
+            Log.info(String.format(CREATE_RESOURCE_GROUP, webAppConfig.getResourceGroup(), webAppConfig.getRegion()));
+            resourceGroup = az.getAzureResourceManager().resourceGroups().define(webAppConfig.getResourceGroup())
+                    .withRegion(webAppConfig.getRegion().getName()).create();
+            Log.info(String.format(CREATE_RESOURCE_GROUP_DONE, webAppConfig.getResourceGroup()));
+        }
         // Get or create App Service Plan
         final IAppServicePlan appServicePlan = az.appServicePlan(webAppConfig.getServicePlanResourceGroup(), webAppConfig.getServicePlanName());
         if (!appServicePlan.exists()) {
+            Log.info(CREATE_APP_SERVICE_PLAN);
             appServicePlan.create()
                     .withName(webAppConfig.getServicePlanName())
-                    .withResourceGroup(resourceGroup.getName())
+                    .withResourceGroup(resourceGroup.name())
                     .withRegion(webAppConfig.getRegion())
                     .withPricingTier(webAppConfig.getPricingTier())
                     .withOperatingSystem(webAppConfig.getRuntime().getOperatingSystem())
                     .commit();
+            Log.info(String.format(CREATE_APP_SERVICE_DONE, appServicePlan.entity().getName()));
         }
-        return webApp.create().withName(webAppConfig.getAppName())
+        Log.info(CREATE_WEBAPP);
+        final IWebApp result = webApp.create().withName(webAppConfig.getAppName())
                 .withResourceGroup(webAppConfig.getResourceGroup())
                 .withPlan(appServicePlan.entity().getId())
                 .withRuntime(webAppConfig.getRuntime())
                 .withDockerConfiguration(webAppConfig.getDockerConfiguration())
                 .withSubscription(webAppConfig.getSubscriptionId())
                 .withAppSettings(webAppConfig.getAppSettings())
                 .commit();
+        Log.info(CREATE_WEB_APP_DONE);
+        return result;
     }
 
     private IWebApp updateWebApp(final IWebApp webApp, final WebAppConfig webAppConfig) throws AzureExecutionException {
         // update app service plan
+        Log.info(UPDATE_WEBAPP);
         final IAppServicePlan currentPlan = webApp.plan();
         final IAppServicePlan targetServicePlan = StringUtils.isEmpty(webAppConfig.getServicePlanName()) ? currentPlan :
                 az.appServicePlan(webAppConfig.getServicePlanResourceGroup(), webAppConfig.getServicePlanName());
         if (!targetServicePlan.exists()) {
-            throw new AzureExecutionException(String.format("App service plan %s was not found in resource group %s",
+            throw new AzureExecutionException(String.format(APP_SERVICE_PLAN_NOT_FOUND,
```

Comment:
add a todo to create the service plan when it doesn't exist

Reply:
Done